==34768== Memcheck, a memory error detector
==34768== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==34768== Using Valgrind-3.22.0 and LibVEX; rerun with -h for copyright info
==34768== Command: ./get_next_line.out 9 r
==34768== Parent PID: 34767
==34768== 
==34768== Invalid read of size 1
==34768==    at 0x10A618: ft_strlen (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A560: handle_string_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A33C: handle_format_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A280: process_format (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A1D2: ft_snprintf (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x109544: tests (src/myunit.c:82)
==34768==    by 0x109A52: main (src/myunit.c:205)
==34768==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==34768== 
==34768== 
==34768== Process terminating with default action of signal 11 (SIGSEGV)
==34768==  Access not within mapped region at address 0x0
==34768==    at 0x10A618: ft_strlen (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A560: handle_string_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A33C: handle_format_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A280: process_format (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A1D2: ft_snprintf (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x109544: tests (src/myunit.c:82)
==34768==    by 0x109A52: main (src/myunit.c:205)
==34768==  If you believe this happened as a result of a stack
==34768==  overflow in your program's main thread (unlikely but
==34768==  possible), you can try to increase the size of the
==34768==  main thread stack using the --main-stacksize= flag.
==34768==  The main thread stack size used in this run was 8388608.
==34768== 
==34768== HEAP SUMMARY:
==34768==     in use at exit: 0 bytes in 0 blocks
==34768==   total heap usage: 0 allocs, 0 frees, 0 bytes allocated
==34768== 
==34768== All heap blocks were freed -- no leaks are possible
==34768== 
==34768== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
==34768== 
==34768== 1 errors in context 1 of 1:
==34768== Invalid read of size 1
==34768==    at 0x10A618: ft_strlen (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A560: handle_string_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A33C: handle_format_specifier (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A280: process_format (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x10A1D2: ft_snprintf (in /home/diego/Documents/Git/mylibs/myunittest/get_next_line.out)
==34768==    by 0x109544: tests (src/myunit.c:82)
==34768==    by 0x109A52: main (src/myunit.c:205)
==34768==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==34768== 
==34768== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
